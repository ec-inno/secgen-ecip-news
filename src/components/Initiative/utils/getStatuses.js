// These are values for filters, not resource path!
// Resource path values are LATEST, ONGOING, ANSWERED, ALL, REFUSED
const getStatuses = t => {
  return [
    { value: 'ALL', label: t('All') },
    { value: 'OPEN', label: t('Open') },
    { value: 'SUCCESSFUL', label: t('Successful') },
    { value: 'ARCHIVED', label: t('Archived') },
  ];
};

export default getStatuses;
