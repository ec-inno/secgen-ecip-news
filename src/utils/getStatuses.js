/**
 * These are values for filters, not resource path!
 * Resource path values are LATEST, ONGOING, ANSWERED, ALL, REFUSED.
 * @see ${api}/register/search/ref/STATUS
 * @param {Function} t Translate function
 * @returns {Array<Object>} Information structured for a select list.
 */
const getStatuses = t => {
  return [
    { value: 'ALL', label: t('All') },
    { value: 'ONGOING', label: t('Ongoing') },
    { value: 'ANSWERED', label: t('Answered') },
  ];
};

export default getStatuses;
