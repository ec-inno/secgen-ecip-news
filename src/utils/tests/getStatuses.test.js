import getStatuses from '../getStatuses';

describe('getStatuses utlity', () => {
  it('returns a list of values', () => {
    const t = s => s; // mocks https://www.i18next.com/overview/api#t
    const statuses = getStatuses(t);
    expect(statuses).toMatchSnapshot();
  });
});
