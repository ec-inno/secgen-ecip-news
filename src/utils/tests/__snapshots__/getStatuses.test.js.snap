// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getStatuses utlity returns a list of values 1`] = `
Array [
  Object {
    "label": "All",
    "value": "ALL",
  },
  Object {
    "label": "Ongoing",
    "value": "ONGOING",
  },
  Object {
    "label": "Answered",
    "value": "ANSWERED",
  },
]
`;
